### PR TITLE
Allow MemSQL to push down string aggregations/joins by forcing a case-sensitive collation

### PR DESCRIPTION
**Pull request by seeday**
Discussion very welcome on this, we can hide it behind a config flag or similar if the previous behavior is still desired.

### CODE REVIEW

**Comment by hashhar** [comment]
The MemSQL tests related to sort order are failing for me with incorrect results (see *CaseSensitive* test methods in TestMemSqlConnectorTest and BaseJdbcConnectorTest).

AFAIK MemSQL (and MySQL) have no collation that matches the codepoint ordering that Trino (and PostgreSQL with `C` collation).

`utf8_bin` doesn't handle non-BMP characters correctly since it's limited to 3 bytes per value.
`utf8mb4_bin` comes close but it doesn't consider trailing spaces when sorting values.

File: plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java [hunk 1]
```diff
@@ -115,11 +115,18 @@ public MemSqlClient(BaseJdbcConfig config, ConnectionFactory connectionFactory,
         this.jsonType = typeManager.getType(new TypeSignature(StandardTypes.JSON));
     }
 
+    @Override
+    public Connection getConnection(ConnectorSession session, JdbcSplit split) throws SQLException
+    {
+        Connection c = super.getConnection(session, split);
+        c.prepareStatement("set collation_connection = 'utf8_bin'").execute();
```
Comment (findepi) [inline plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java:122]: This should be done in connection factory.
**For example**, when DELETE is delegated to the database, same settings should be applied.

**Comment by hashhar** [comment]
> It looks like the Postgres `C` collation is a simple byte comparison, which seems like it would map to MemSQL's `binary` collation. However, that also seems like it would break pushdown of certain string manipulations (`upper`/`lower`/etc.) if it was set on the entire connection (though I don't know if Trino does that kind of push down).

MemSQL doesn't have a `binary` collation for types other than BINARY and VARBINARY. It has `utf8_bin` (and similar) for CHAR, VARCHAR and other text types. None of the collations that MemSQL ships with behave like the C collation.
MySQL 8 added a new collation that seems like what is needed but MemSQL doesn't have it.

> Hmm, perhaps I relied on the CI tests too much. It seemed like it ran and passed the tests there, does the CI skip some of the MemSQL tests?

CI for MemSQL doesn't run for pull-requests created from a fork (to avoid leaking GitHub secrets which include a MemSQL license). You can run the test class locally since it's all dockerised.

**Comment by mosabua** [comment]
:wave: @seeday  - this PR has become inactive. If you're still interested in working on it, please let us know.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by colebow** [comment]
Closing this one out due to inactivity, but please reopen if you would like to pick this back up.

**Pull request closed without merging**